corehq/apps/reports_core/filters.py

```diff
@@ -118,8 +119,10 @@ def value(self, **kwargs):
         date_range_inclusive = kwargs.get('date_range_inclusive', True)
 
         def date_or_nothing(param):
-            return datetime.strptime(param, "%Y-%m-%d") \
-                if param else None
+            if param:
+                return datetime.combine(iso_string_to_date(param), time())
```

Comment:
I ended up doing this a lot as the translation from using strptime directly. I considered writing a helper function ("date_to_datetime_with_zeroed_time") but that actually seemed wrong. I wanted to do it because I wanted doing this to be easy, for the sake of the refactor. But it *shouldn't* be easy, it's usually a dumb, meaningless thing to do and in many contexts I've come across it was done more or less as a mistake (or at the very least, carelessly, because `datetime.strptime` made it so easy)